### DISCUSSION TITLE
Button font size on backgroundSelectedColor

### DISCUSSION THREAD

**Discussion by caspahouzer** [appcelerator/titanium_mobile#12668] [Q&A] [answered]
If I run this code on iOS...
```
<Alloy>
	<Window>
		<View layout="vertical" height="Ti.UI.SIZE">
			<Button backgroundColor="#8fb63e" top="10" height="40" width="300">This is a test</Button>
			<Button backgroundColor="#8fb63e" top="10" height="40" width="300" backgroundSelectedColor="red">This is a test</Button>
		</View>
	</Window>
</Alloy>
```
the fontsize on the second button differs to the first as you can see in the image

<img width="375" alt="Captura de Pantalla 2021-03-29 a la(s) 1 05 19 a m" src="https://user-images.githubusercontent.com/1126641/112801861-b1957b80-9071-11eb-9c0f-e0b5d90ecd50.png">

I filed a [Jira issue](https://jira.appcelerator.org/browse/AC-6689) for that. But maybe someone has an idea how to fix that quick

**Comment by drauggres** [comment]
You should have open an [issue](https://github.com/appcelerator/titanium_mobile/issues), not a discussion.
Also, attached app sample is usually in "classic Titanium", not Alloy (unless it is Alloy related issue).

**Reply by Topener** [reply]
@drauggres a discussion is fine for a question how to avoid a known bug though, this is not a bug report :)

**Reply by macCesar** [reply]
@drauggres ... If EVERYBODY ( and in EVERY article ) they recommend working with Allow rather than Classic Ti...

Why should we have to submit sample code in Classic Ti ??? 🤷‍♂️

**Reply by m1ga** [reply]
because it is a lot easier/quicker to test! You can just paste the code and don't have to create a controller, view, style file. E.g. in this example you'll need a controller file with `$.index.open()`. Style is inline so that's fine.

**Reply by Topener** [reply]
Also, Alloy is compiled down to Classic in the end. Your compiled code will not contain xml or tss, just the classic .js files.

**Reply by macCesar** [reply]
Ok I grant both of you your answers...

But... Can't the original code, the description and maybe some previews be enough to hint you about the problem without running it in a new proyect?? 

That is how we used to review and debug code in college!!! 🤣🤣🤣

Reading and analyzing each line of code, In order to understand the logic and procedures!! 🤯🤯🤯

Anyway.. thanks for all your help!!! 👍🏻👍🏻

**Reply by macCesar** [reply]
Also.. if I have to rewrite the original code to Classic Ti...

It may not represent the original problem... or it may be incomplete and lead to different results.

**Reply by drauggres** [reply]
> Also.. if I have to rewrite the original code to Classic Ti...
> 
> It may not represent the original problem... or it may be incomplete and lead to different results.

Then that is an Alloy problem. That is the point: to report a bug in the SDK, you must provide code that has minimum external influence, i.e. classic Ti.

This is the way.

**Reply by macCesar** [reply]
I remember a few years ago, I was helping a junior developer.. 

And he wanted to extract a piece code that he thought was causing the problem in a larger function.

I told him that I wanted to see the whole function.. It was only like 30 or 40 lines of code.

Anyway..  just by doing a quick scan of the code, I toll him "Your are missing a semicolon in line XX". 

After adding the semicolon and running the function, everything was working correctly...

He was struggling for almost an hour.. and just for missing a semicolon!! 

It was in PHP and he didn't understand the error message.

We've all been there!!! 🤦‍♂️🤦‍♂️🤦‍♂️

My point is that by showing the ORIGINAL code.. people with more experience or that have encountered the same problem can help out easily!!!

**Reply by m1ga** [reply]
I think the point here is just: the problem is not in the code but inside the SDK. So if you paste the responsible JAVA lines/code it the same as you are saying. But here the code is just to show the actual bug. And the quicker people can run the code to reproduce the error (that is somewhere inside the SDK/JAVA code) its easier to fix it :wink: 
As it was mentioned in the original reply: if it is an Alloy error its fine, if not: the appc devs can test it with Alloy code too but the setup is a bit more and perhaps you miss some styles or code in the controller that is missing in the Alloy view example.

Anyway: there is a bug somewhere here: https://github.com/appcelerator/titanium_mobile/blob/5c5efc067ba1f1ba3a3bfce03ade762634e9465a/android/titanium/src/java/org/appcelerator/titanium/view/TiUIView.java#L1413-L1427 / in `applyCustomBackground` I guess :smile:

**Reply by drauggres** [reply]
> My point is that by showing the ORIGINAL code.. people with more experience or that have encountered the same problem can help out easily!!!

That is fine if don't know what is going on.

But then we are talking about bug report you should provide minimal reproducible example (aka [MWE](https://en.wikipedia.org/wiki/Minimal_working_example)). Especially with this project: there are too few developers and too much work for them. By providing code in Ti Classic you are really helping yourself to get fix quicker.

**Reply by macCesar** [reply]
@Topener

Maybe you need to add more categories so people can place their questions in the proper place ??

Alloy, Classic or Native Code ??

Just a thought!! 🙂

**Reply by Topener** [reply]
@macCesar that's absolutely something we can look into, but I think the volume of questions isn't big enough to split it up just yet

**Comment by Topener** [comment, marked as answer]
Does just setting a font work? https://titaniumsdk.com/api/titanium/ui/button.html#font

**Reply by caspahouzer** [reply]
Yes. If I set fontSize for the button, it works as expected

**Reply by macCesar** [reply]
But the question remains... Why the change in font size and color just by setting `selectedBackgroundColor`?

**Reply by Topener** [reply]
That is a good question, but that is a bugreport worthy! This question was purely to find a workaround for the bug

**Reply by caspahouzer** [reply]
https://jira.appcelerator.org/plugins/servlet/mobile#issue/AC-6689 😊
